src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -355,20 +366,23 @@ export class HttpConnection implements IConnection {
                 transportExceptions.push(`${endpoint.transport} failed: ${transportOrError}`);
             } else if (this.isITransport(transportOrError)) {
                 this.transport = transportOrError;
-                if (!negotiateResponse.connectionId) {
+                if (!negotiateResponse.connectionToken) {
                     try {
                         negotiateResponse = await this.getNegotiationResponse(url);
+                        this.connectionId = negotiateResponse.connectionId;
```

Comment:
Would it be possible to wait to set `this.connectionId` until after startTransport, so you don't need to unset it in the catch? Or would that be too late?